pep_sphinx_extensions/pep_zero_generator/parser.py

```diff
@@ -0,0 +1,165 @@
+
+if TYPE_CHECKING:
+    from pep_sphinx_extensions.pep_zero_generator.author import Author
+
+
+class PEP:
+    """Representation of PEPs.
+
+    Attributes:
+        number : PEP number.
+        title : PEP title.
+        pep_type : The type of PEP.  Can only be one of the values from TYPE_VALUES.
+        status : The PEP's status.  Value must be found in STATUS_VALUES.
+        authors : A list of the authors.
+
+    """
+
+    # The required RFC 822 headers for all PEPs.
+    required_headers = {"PEP", "Title", "Author", "Status", "Type", "Created"}
+
+    def __init__(self, filename: Path, authors_overrides: dict):
+        """Init object from an open PEP file object.
+
+        pep_file is full text of the PEP file, filename is path of the PEP file, author_lookup is author exceptions file
+
+        """
+        self.filename: Path = filename
+
+        # Parse the headers.
+        pep_text = filename.read_text(encoding="utf-8")
+        metadata = HeaderParser().parsestr(pep_text)
+        required_header_misses = PEP.required_headers - set(metadata.keys())
+        if required_header_misses:
+            _raise_pep_error(self, f"PEP is missing required headers {required_header_misses}")
+
+        try:
+            self.number = int(metadata["PEP"])
+        except ValueError:
+            _raise_pep_error(self, "PEP number isn't an integer")
+
+        # Check PEP number matches filename
+        if self.number != int(filename.stem[4:]):
+            _raise_pep_error(self, f"PEP number does not match file name ({filename})", pep_num=True)
+
+        # Title
+        self.title: str = metadata["Title"]
+
+        # Type
+        self.pep_type: str = metadata["Type"]
+        if self.pep_type not in TYPE_VALUES:
+            _raise_pep_error(self, f"{self.pep_type} is not a valid Type value", pep_num=True)
+
+        # Status
+        status = metadata["Status"]
+        if status in SPECIAL_STATUSES:
+            status = SPECIAL_STATUSES[status]
+        if status not in STATUS_VALUES:
+            _raise_pep_error(self, f"{status} is not a valid Status value", pep_num=True)
+
+        # Special case for Active PEPs.
+        if status == STATUS_ACTIVE and self.pep_type not in ACTIVE_ALLOWED:
+            msg = "Only Process and Informational PEPs may have an Active status"
+            _raise_pep_error(self, msg, pep_num=True)
+
+        # Special case for Provisional PEPs.
+        if status == STATUS_PROVISIONAL and self.pep_type != TYPE_STANDARDS:
+            msg = "Only Standards Track PEPs may have a Provisional status"
+            _raise_pep_error(self, msg, pep_num=True)
+        self.status: str = status
+
+        # Parse PEP authors
+        self.authors: list[Author] = _parse_authors(self, metadata["Author"], authors_overrides)
+
+    def __repr__(self) -> str:
+        return f"<PEP {self.number:0>4} - {self.title}>"
+
+    def __lt__(self, other: PEP) -> bool:
```

Comment:
You may want to implement `__eq__` as well.

Reply:
I think it would only be used if two PEPs had the same number, which theoretically should never happen? Let me know if you've a strong feeling on this one but I would probably hold off for clarity of expression. 

I was half thinking of making the PEP class a NamedTuple/Dataclass like Author, but held off as I'd need to override too much (sorting, custom details method, custom init) that it wasn't worth it

Reply:
No strong feeling, just that is a bit odd to define order but not direct equality, but I agree that it doesn't matter

Reply:
done this too as why not